core/src/test/java/org/bitcoinj/base/Bech32Test.java

```diff
@@ -71,12 +71,18 @@ private void valid(String valid) {
     public void invalid_bech32() {
         for (String invalid : INVALID_BECH32)
             invalid(invalid);
+        // Valid BECH32M codes are invalid BECH32
```

Comment:
I think this is a false assumption. It assumes that the decoder would decode only one of the two formats "at a time", but it decodes both at the same time.

Reply:
This makes sense. I'll make a PR with some JavaDoc for `decode` to make this more clear.

We might want to add an optional encoding parameter to `decode` so it will fail if string is in the wrong encoding. (Similar to what we do with the network parameter for addresses.) This could be used by the unit tests to verify that the distinctions between Bech32 and Bech32m are handled properly.